### PR TITLE
Alihamud live code

### PR DESCRIPTION
**Pull request by ali-hamud**
Live Demo with visible code.

### CODE REVIEW

File: demo/code-demo/report.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    </div>
+    
+    <div id="embed-and-interact-panel">
+        <div id="right-pane" class="halfWidth left">
+        </div>
+
+        <div id="left-pane" class="halfWidth right">
+            <div id="embedArea">
+                <h3>Embedded Report Area</h3>
+                <h5>The following div id is <b>reportContainer</b>. In code, we embed a report to it.</h5>
```
Comment (ranbreuer) [inline demo/code-demo/report.html]: I think we need to go through text scrubbing for the texts here..

File: demo/code-demo/scripts/codesamples.js [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    
+    var reportContainer = document.getElementById('reportContainer');
+    powerbi.embed(reportContainer, embedConfiguration);
+}
+
+// ---- Report Operations ----------------------------------------------------
+
+ function _Report_GetId() {
+    report = powerbi.embeds[0];
+    $('#result').html(report.getId());
```
Comment (noanu) [inline demo/code-demo/scripts/codesamples.js]: Is"report" always defined? is there a chance for referencing a null here?

File: demo/code-demo/scripts/session_utils.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+function SetSessions(accessToken, embedReportUrl, embedReprotId) {
+    SetSession("accessToken", accessToken);
+    SetSession("embedUrl", embedReportUrl);
+    SetSession("embedId", embedReprotId);
+}
+
+function UpdateSessions() {
+    var txtAccessToken = $('#txtAccessToken').val();
+    if (txtAccessToken)
```
Comment (noanu) [inline demo/code-demo/scripts/session_utils.js]: what happens if these strings are empty strings? do you want to update the session with them anyway?

File: demo/code-demo/scripts/session_utils.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    var txtEmbedReportId = $('#txtEmbedReportId').val();
+    if (txtEmbedReportId)
+    {
+        SetSession("embedId", txtEmbedReportId);
+    }
+}
+
+function SetAccessTokenFromSession() {
```
Comment (noanu) [inline demo/code-demo/scripts/session_utils.js]: the session should always be updated with correct values and is responsible for checking for nulls when it is updates, so is it neccessary do check it also here?

File: demo/code-demo/scripts/step_authorize.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+var embedWithSpecificReportDivExpanded = false;
+
+function ExpandEmbedWithSpecificReportDiv() {
```
Comment (noanu) [inline demo/code-demo/scripts/step_authorize.js]: shouldn't this function name be somethings with "toggle" (that was it does doesn't it?)

File: demo/code-demo/scripts/utils.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        alert("You must specify an embed url.");
+        return false;
+    }
+    var id = null;
+    var parts = embedUrl.split("reportId=");
+    if (parts && parts.length > 0)
+    {
+        var guidParts = parts[parts.length  -1].split("&");
+        id = guidParts[0];
```
Comment (noanu) [inline demo/code-demo/scripts/utils.js]: is there a chance "guidParts" is null?

File: demo/code-demo/scripts/session_utils.js (hunk 3 above)
Comment (ali-hamud) [inline demo/code-demo/scripts/session_utils.js]: Yes

File: demo/code-demo/scripts/codesamples.js (hunk 2 above)
Comment (ali-hamud) [inline demo/code-demo/scripts/codesamples.js]: Actually report can be null if there is no embeds.

Comment (ali-hamud) [inline demo/code-demo/scripts/codesamples.js]: But the code will not be that clear.
this is a sample code for developers, they should add the check after copying.

**Pull request merged**